### ISSUE TITLE
Leveraging Blender 4 and Godot 4 enhancements for swift game-ready character creation in MPFB

### ISSUE THREAD

**Issue by ashledombos** [makehumancommunity/mpfb2#152]
**Is your feature request related to a problem? Please describe.**
The feature request aims to capitalize on the recent and upcoming enhancements in Blender 4 and Godot 4 to establish MPFB as a rapid character creation platform for Godot.

**Describe the solution you'd like**
1. **Exportation Directe**:
   - Develop a direct export feature from MPFB to Godot, utilizing the reliable GLTF configurations and possibly creating a dedicated export/import plugin for a seamless transition of assets.

2. **Game Ready Optimization**:
   - Implement tools within MPFB for optimizing models to be game-ready. This includes polygon reduction, texture optimization, rig and animation adjustments to ensure maximum performance in Godot.

3. **Retargeting Improvements**:
   - Explore the new retargeting system in Godot 4.0, particularly the SkeletonProfileHumanoid preset, to streamline the application of animations between different models. Enhance BVH retargeting workflows from MPFB to Godot, reducing the effort required to share animations.

4. **Documentation and Tutorials**:
   - Offer clear documentation and tutorials on the improved workflow between MPFB, Blender 4, and Godot 4. This will help the community to effectively navigate and utilize the integrated systems for rapid character creation.

5. **Collaboration with Godot Developers**:
   - Initiate dialogue with the Godot community to explore best practices and technical standards for facilitating integration and collaboration between MPFB and Godot.

6. **Addressing Export Issues**:
   - Address reported issues with the gltf/glb exporter concerning the support of Blender’s visual materials to ensure accurate and efficient asset export to Godot.

**Describe alternatives you've considered**
There are certainly other less or more known platforms, both proprietary such as Unity and Unreal Engine, and open-source ones like Löve, which offer robust game development environments. But I‘m only familiar with Godot…

**Additional context**
With the evolving enhancements in Blender 4 and Godot 4, there’s a significant opportunity for MPFB to evolve into a robust, user-friendly platform for quickly creating and exporting characters to Godot. Engaging with the Godot community and keeping abreast of the technical advancements in both Blender and Godot could position MPFB as a valuable tool for game developers and artists in the Godot ecosystem.

**Comment by joepal1976** [comment]
Yes, in the best of worlds, there should be good Godot support. There has been some discussions on the forum regarding exporting from MakeHuman to Godot, but I haven't really looked into it on the mpfb side. 

Some parts are game engine independent, such as the polygon reduction and the texture parts. And some parts of that are mostly a question of documentation.

I could really use help regarding the Godot-specific parts though, as I have never used it personally. For Unity, I have a grasp of the basics and can test what consequences design choices in mpfb have.

**Comment by joepal1976** [comment]
In related news, there seems to be a godot side of things in progress: http://www.makehumancommunity.org/forum/viewtopic.php?f=23&t=23056

**Comment by IvailoBurov** [comment]
This is a project in development to create MPFB2 characters directly in Godot:
https://github.com/NitroxNova/Humanizer

**Comment by spamrakuen** [comment]
I'm trying to use MPFB2 for Godot (currently in my workflow I'm still using MakeHuman 1.2).
Until now, what I've missed most is a control rig for the "game engine" rig, that's why I've created a simple one for my need.
But something like the old MHX Rig or Rigify support for the "game engine" rig would be nice.

**Comment by SanjoSolutions** [comment]
I currently use [Game Rig Tools (my extended version)](https://github.com/SanjoSolutions/game-rig-tools) to prepare MPFB characters with Rigify rig for export to Godot. I let Godot import the Blender file directly. I also have a [few modifications for MPFB](https://github.com/SanjoSolutions/mpfb2/tree/game-rig-tools-support) for improved support for Rigify/Game Rig pairs.

You can merge/cherry-pick the changes if you like.
